docs/proposals/namespace-deletion-protection/README.md

```diff
@@ -0,0 +1,86 @@
+
+### Goals
+
+- Add or remove protection labels by karmadactl.
+- Prevent the user's deletion operation.
+
+## Proposal
+
+Protection Label Name: `karmada.io/deletion-protected`
+
+We can add a label `karmada.io/deletion-protected=true` to the namespace resource that needs protection. When a resource with this label is being deleted, we need to interrupt the user's operation request.
+
+### User Story
+
+As a user, I want to protect namespace resources, so that they won't be accidentally deleted.
+
+## Design Details
+
+### Add protect label
+
+There are two ways to add or remove protection labels here.
+
+#### Add the label
+`kubectl label <resource-type> <resource-name> karmada.io/deletion-protected=true`
+
+`karmadactl protect <resource-type> <resource-name>`
+
+#### Delete the label
+`kubectl label <resource-type> <resource-name> karmada.io/deletion-protected=false`
+
+`karmadactl protect remove <resource-type> <resource-name>`
+
+### Prevent deletion operation
+
+For preventing deletion operations, we can achieve it through an Admission webhook. When a user attempts to delete, we need to first check if the label has a value of "true". If it does, we will interrupt the user's deletion operation.
+![admission_webhook.png](statics/admission_webhook.png)
+
+### Test Plan
+
+#### UT
+
+Test cover the new methods.
+
+#### E2E
+
+| Test Case                                                  | Case Description                                                                                              |
+|------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------|
+| Delete Namespace with no label                             | Verify if the system allows deletion of a namespace with no label.                                            |
+| Delete Namespace with protect label                        | Test if the system allows deletion of a namespace that has a 'protect' label.                                 |
+| Delete Namespace after protect label has been set to false | Check the functionality of the system in deleting a namespace once its 'protect' label has been set to false. |
+
+## Alternatives
+
+### Why choose Label over Annotations to tag resources?
+
+Annotations are typically used to store metadata that is not used for identification and selection, such as descriptive information or the status information of management tools. In this case, labels like "cannot be deleted" or "protected status" are not suitable as annotations.
+
+### What will happen If user want to delete namespace by `--force`?
+
+I believe we shouldn't block this operation. Our goal is to prevent users from accidentally deleting resources, but if a user uses the `--force` parameter, we should assume that they know what they are doing.
```

Comment:
If a user uses `--force` to perform a force delete, can the webhook detect this request and allow it?

Reply:
This is a good question. We might not be able to distinguish between force delete and normal delete.